src/context/AppReducer.js

```diff
@@ -10,6 +10,18 @@ function AppReducer(state, action) {
         ...state,
         cartItems: action.payload,
       };
+    case "UPDATE_ITEM_AMOUNT": {
```

Comment:
Veikia ir šitaip, bet aš kitaip daryčiau nes iš pradžių kilo įtarimas kad gal šitas būdas mutatina state tiesiogiai 🤔

Čia šiaip diskusijai, jei taip kaip yra labiau patinka - galim palikt, aš ne profė 😀

` case "UPDATE_ITEM_AMOUNT": {
      return {
        ...state,
        cartItems: handleCartItemAmountChange(state.cartItems, action.payload),
      };`


```

function handleCartItemAmountChange(cartItems, payload) {
  const index = cartItems.findIndex((item) => item.id === payload.id);
  const item = cartItems[index];

  const changedItem = {
    ...item,
    amount: payload.amount
  };

  return [...cartItems.slice(0, index), changedItem, ...cartItems.slice(index + 1)];
}

```

Reply:
Bet kadangi veikia kaip yra galim palikt, nesitikėjau iš tikro kad veiks, matyt nemoku kažko dar 😂

Reply:
Jo :D siaip cia neveikia :D kitu pr istaisiau klaida. Jis veikia cia del to jog sitoj versijoj yra neoptimizuotas pushas i duombaze, ir kazkuriuo metu pushina su pakeista objekto rodykle. 
![image](https://user-images.githubusercontent.com/57706951/112676574-441afc80-8e71-11eb-94b3-24d199ffe7ca.png)

Reply:
va taip arba galima ieskoti indekso ir pakeisti objekta vietoj for loopo

Reply:
Su #10 ?

Reply:
Jo man mano variantas labiau patinka vien dėl to kad nėra loop ir break xD

Reply:
aha. sita approvink #10 padarysiu index

Reply:
Nu bet svarbu veikia teisingai, kam kaip patinka taip tegu rašo

Reply:
aha. sita approvink. #10 padarysiu index

Reply:
:D

Reply:
a dot

Reply:
ir dar kitam merge konfliktai yr